index.bs

```diff
@@ -3296,6 +3300,270 @@ failed</dfn> steps given |context| and |navigation status|:
+the request, relative to <code>requestTime</code>.
+
+TODO: Add service worker fields
+
+#### The network.Header type #### {#type-network-Header}
+
+[=Remote end definition=] and [=local end definition=]
+
+<pre class="cddl local-cddl">
+NetworkHeader = {
+    name: text,
+    ? value: text,
+    ? binaryValue: [ uint ]
+};
+</pre>
+
+The <code>NetworkHeader</code> type represents a single request header.
+
+If the header value can be represented as a UTF-8 encoded string, the
+<code>value</code> field will be present. Otherwise the <code>binaryValue</code>
+field will be present and consist of an array of integers representing the bytes
+of the header.
+
+#### The network.Initiator type #### {#type-network-Initiator}
+
+[=Remote end definition=] and [=local end definition=]
+
+<pre class="cddl local-cddl">
+NetworkInitiator = {
+    type: "parser" / "script" / "preflight" / "other",
+    ?columnNumber: uint,
+    ?lineNumber: uint,
+    ?stackTrace: StackTrace,
+    ?request: NetworkRequest
+};
+</pre>
+
+The <code>NetworkInitiatior</code> type represents the source of a network request.
+
+TODO: Align more closely with Fetch here?
+
+#### The network.Request type #### {#type-network-RequestId}
+
+[=Remote end definition=] and [=local end definition=]
+
+<pre class="cddl local-cddl remote-cddl">
+NetworkRequest = text;
+</pre>
+
+Each network request has an associated <dfn export>request id</dfn>, which is a string
+uniquely identifying that request.
+
+#### The network.RequestInfo type #### {#type-network-RequestInfo}
+
+[=Remote end definition=] and [=local end definition=]
+
+<pre class="cddl local-cddl">
+NetworkRequestInfo = {
+    url: text,
+    method: text,
+    headers: [ *NetworkHeader ]
+    ?body: text,
+    bodySize: uint,
+    headersSize: uint
+};
+</pre>
+
+The <code>NetworkRequestInfo</code> type represents an ongoing network request.
+
+TODO: Boduy is actually bytes, not clear how to handle it as text
+
+#### The network.ResponseData type #### {#type-network-ResponseData}
+
+[=Remote end definition=] and [=local end definition=]
+
+<pre class="cddl local-cddl">
+NetworkResponseData = {
+    url: text,
+    protocol: text,
+    status: unit,
+    statusText: text,
+    headers: [ *NetworkHeader ],
+    cookies: [ *NetworkCookie ],
+    mimeType: text,
+    bytesReceived: uint,
+    timings: NetworkFetchTimingInfo,
+};
+</pre>
+
+### Events ### {#module-network-events}
+
+#### The network.beforeRequestSent Event #### {#event-network-beforeSendRequest}
+
+<dl>
+   <dt>Event Type</dt>
+   <dd>
+      <pre class="cddl local-cddl">
+        NetworkBeforeRequestSentEvent = {
+         method: "network.beforeRequestSent",
+         params: NetworkBeforeRequestSentParams
+       }
+
+       NetworkBeforeRequestSentParams = {
+        request: Request,
+        navigation: Navigation / null,
+        context: BrowsingContext / null,
+        requestData: RequestData,
```

Comment:
There's no definition for (Network?)RequestData but we have an unused NetworkRequestInfo. If that's what this is supposed to be, then we should have request cookies as well in NetworkRequestInfo?

Reply:
Thanks! I renamed `NetworkRequestInfo` to `NetworkRequestData` and added the cookies. I also cleaned up a few other misnamed things.